pkg/controller/group/group.go

```diff
@@ -181,10 +181,19 @@ func (p *gController) DestroyGroup(gid group.ID) error {
 		if err != nil {
 			return err
 		}
-
-		for _, desc := range descriptions {
+		// Destroy self last
```

Comment:
The code that orders the slice of instances to destroy will make sure the self node is always the last entry in the slice https://github.com/docker/infrakit/blob/master/pkg/controller/group/state.go#L129

So is this still necessary?   If anything we should remove the policy checks and just make sure the self node is always last from the sort.

Reply:
So that sorting is **only** done in the **rolling update** flow: https://github.com/docker/infrakit/blob/master/pkg/controller/group/rollingupdate.go#L252

The **group destroy** flow being updated in this PR.

What do you think about updating the generic `scaled.List` to apply the same sort? https://github.com/docker/infrakit/blob/master/pkg/controller/group/scaled.go#L137

The benefit of this is that the current leader will always be returned last for any `List` operation and no addition work is needed to ensure that "self" is done last.